net/mptcp/mptcp_ctrl.c

```diff
@@ -1104,7 +1104,12 @@ static int mptcp_alloc_mpcb(struct sock *meta_sk, __u64 remote_key,
 	}
 #endif
 
-	meta_tp->mptcp = NULL;
+	meta_tp->mptcp = kmem_cache_zalloc(mptcp_sock_cache, GFP_ATOMIC);
```

Comment:
Why is this needed?

Reply:
I see.

The thing is that for the meta-socket, we store scheduling information in the mpcb. Thus, in blestsched_modify_lambda you should rather get the priv from there instead of using mptcp_tcp_sock.

Makes sense?

Reply:
I will rework the code in `mptcp_blest.c` to get the priv from the mpcb properly and remove the changes in `mptcp_ctrl.c`, because this fix should then no longer be needed.